components/server/src/api/workspace-service-api.ts

```diff
@@ -92,4 +99,79 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.pagination.total = resultTotal;
         return response;
     }
+
+    async createAndStartWorkspace(req: CreateAndStartWorkspaceRequest): Promise<CreateAndStartWorkspaceResponse> {
+        // We rely on FGA to do the permission checking
+        if (req.source?.case !== "contextUrl") {
+            throw new ApplicationError(ErrorCodes.UNIMPLEMENTED, "not implemented");
+        }
+        if (!req.organizationId || !uuidValidate(req.organizationId)) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "organizationId is required");
+        }
+        if (!req.editor) {
```

Comment:
I don't think it should be required. We should derive from settings, most likely implementation already does it.

Reply:
I know that Dashboard provides it always, but i don't think it makes sense to require on API surface.

Reply:
It could be a follwo up issue in API hardening milestone if hard